src/android/io/cozy/plugins/listlibraryitems/ListLibraryItems.java

```diff
@@ -325,13 +324,11 @@ protected Integer doInBackground(UploadFilePayload... uploadFilePayloads) {
                 OutputStream out = huc.getOutputStream();
                 buffer_size = Math.min(bytes_available, MAX_BUFFER_SZ);
                 buffer = new byte[buffer_size];
-                bytes_read = fis.read(buffer, 0, buffer_size);
                 total_bytes = bytes_read;
-                while (bytes_read > 0) {
+                while ((bytes_read = fis.read(buffer, 0, buffer_size)) != -1) {
```

Comment:
In case of 0 length file or other strange behaviour, perhaps it's better to "<= 0" instead of "!=1"

Reply:
Regarding the `FileInputStream.read()` documentation:
> the next byte of data, or -1 if the end of the file is reached.